### PR TITLE
Supporting 15 more resource types

### PR DESCRIPTION
**Pull request by samuelgautier**

### CODE REVIEW

File: azurecaf/models.go [hunk 1]
```diff
@@ -12,14 +12,16 @@ const (
...
-	invsqldb    string = "[<>*%&:\\/?]" //sql db invalid character
+	alphanum     string = "[^0-9A-Za-z]"
+	alphanumh    string = "[^0-9A-Za-z-]"
+	alphanumhp  string = "[^0-9A-Za-z\\.\\-]"
+	alphanumu    string = "[^0-9A-Za-z_]"
+	alphanumhu   string = "[^0-9A-Za-z_-]"
+	alphanumhup  string = "[^0-9A-Za-z_.-]"
+	alphanumhups string = "[^0-9A-Za-z_\\.\\-\\/]"
+	unicode      string = `[^-\w\._\(\)]`
+	invappi      string = "[%&\\?/]"     //appinisghts invalid character
```
Comment (patpicos) [inline azurecaf/models.go:23]: typo in comment

File: README.md [hunk 2]
```diff
@@ -80,25 +80,40 @@ Current prototype supports:
...
 | AKS Node Pool Linux                 | aksnpl                      | aks_node_pool_linux                     |
 | AKS Node Pool Windows               | aksnpw                      | aks_node_pool_windows                   |
 | Azure Site Recovery                 | asr                         | azurerm_recovery_services_vault         |
+| Azure Availability Set              | avail                       | azurerm_availability_set                |
+| Azure Vpn Connection                | cn                          | azurerm_vpn_connection                  |
 | Azure Event Hubs                    | evh                         | azurerm_eventhub_namespace              |
 | generic                             | gen                         | generic                                 |
 | Azure Key Vault                     | kv                          | azurerm_key_vault                       |
 | Azure Monitor Log Analytics         | la                          | azurerm_log_analytics_workspace         |
+| Azure Load Balancer (External)      | lbe                         | azurerm_load_balancer_external          |
```
Comment (patpicos) [inline README.md:95]: ive often seen ilb vs. elb for internal/external. I know its not lb* prefix, but its fairly common.

File: azurecaf/models.go [hunk 3]
```diff
@@ -61,21 +63,36 @@ var Resources = map[string]ResourceStructure{
 	"app":    {"web app", "app", 2, 60, false, alphanumh, "^[0-9A-Za-z][0-9A-Za-z-]{0,58}[0-9a-zA-Z]$"},
 	"appi":   {"application insights", "appi", 1, 260, false, invappi, "^[^%&\\?/. ][^%&\\?/]{0,258}[^%&\\?/. ]$"},
 	"ase":    {"app service environment", "ase", 2, 37, false, alphanumh, "^[0-9A-Za-z-]{2,37}$"},
+	"asg":    {"application security group", "asg", 1, 80, false, alphanumhup, "^[0-9a-zA-Z][0-9A-Za-z_.-]{0,78}[0-9a-zA-Z_]$"},
```
Comment (nyuen) [inline azurecaf/models.go:66]: Thanks for the contribution! I've modified the structure in the latest release, could you add update merge the new changes and push?- I'll then integrate the changes

**Review by nyuen — CHANGES_REQUESTED** [review]

File: azurecaf/models.go (hunk 3 above)
Comment (samuelgautier) [reply azurecaf/models.go:66]: Where would you like that i declare the new resource types in the model ? (models.go or in the json files)

File: .gitignore [hunk 4]
```diff
@@ -10,3 +10,4 @@ terraform
 **/.terraform.tfstate.lock.info
 **/~*.*
 **/*.log
+go.mod
```
Comment (Nepomuceno) [inline .gitignore:13]: go mod it is normally included in the prs like a packages

**Comment by arnaudlh** [comment]
hi @samuelgautier those should have been addressed in #38 please let us know if your are still facing the issue

**Pull request closed without merging**